zulipterminal/ui_tools/boxes.py

```diff
@@ -97,6 +97,15 @@ def __init__(self, view: Any) -> None:
         # Prioritizes autocomplete in message body
         self.recipient_user_ids: List[int]
 
+        # Marks if the write box formed is involves [EDIT_MESSAGE: e] key
+        # Only used for private messages(for now)
+        self.edit_keypressed: bool
+        self.edit_keypressed = False
+
+        # Marks if the write box formed is involves [REPLY_MESSAGE: r] key
+        # Only used for private messages(for now)
+        self.reply_keypressed: bool
+        self.reply_keypressed = False
```

Comment:
Why are we tracking if it's a reply? Ah, this is since private edits are triggered by reusing the reply logic?

Reply:
Yes, the keypress of `EDIT_MESSAGE` on private messages works on calling the keypress for `REPLY_MESSAGE` using `self.keypress(size, primary_key_for_command("REPLY_MESSAGE"))`.That's why reply_keypressed is used to differentiate between cases .